build-system/tasks/extension-helpers.js

```diff
@@ -561,18 +563,54 @@ async function buildExtensionJs(path, name, version, latestVersion, options) {
-    const file = `dist/v0/amp-script-worker-${version}`;
-    // The "js" output is minified and transpiled to ES5.
-    fs.copyFileSync(dir + 'worker.js', `${file}.js`);
-    // The "mjs" output is unminified ES6 and has debugging flags enabled.
-    fs.copyFileSync(dir + 'worker.mjs', `${file}.max.js`);
-
-    // Same as above but for the nodom worker variant.
-    const noDomFile = `dist/v0/amp-script-worker-nodom-${version}`;
-    fs.copyFileSync(dir + 'worker.nodom.js', `${noDomFile}.js`);
-    fs.copyFileSync(dir + 'worker.nodom.mjs', `${noDomFile}.max.js`);
+    const workerDomDir = 'node_modules/@ampproject/worker-dom';
+    const newDirExists = fs.existsSync(`${workerDomDir}/dist/amp-production`);
+    const targetDir = 'dist/v0';
+
+    log('Copying required @ampproject/worker-dom files.');
+
+    if (newDirExists) {
+      const dir = `${workerDomDir}/dist`;
+      const workerFilesToDeploy = await globby(
+        `${dir}/amp-production/worker/*`,
+        `${dir}/amp-debug/worker/*`
+      );
+      for (const filepath of workerFilesToDeploy) {
+        const maxFile = /amp-debug/.test(filepath);
+        const basename = pathmod.basename(filepath);
+        // Modify the current name to be either amp-script-worker or
+        // amp-script-worker-nodom.
+        const newBasename = basename
+          .replace(/^worker/, 'amp-script-worker')
+          .replace(/\.nodom/, '-nodom');
+        const nameParts = newBasename.split('.');
+        // Modify the basename part to have the version and the max.
+        nameParts[0] = nameParts[0] + `-${version}${maxFile ? '.max' : ''}`;
+        const newFilename = nameParts.join('.');
+        await fs.copy(filepath, `${targetDir}/${newFilename}`);
+      }
+      // TODO(#30142, erwinm): remove else block once
+      // https://github.com/ampproject/worker-dom/pull/929 is the new version
+      // installed by AMP.
+    } else {
+      const dir = `${workerDomDir}/dist/amp/worker/`;
+      const file = `${targetDir}/amp-script-worker-${version}`;
+      // The "js" output is minified and transpiled to ES5.
+      await fs.copy(`${dir}/worker.js`, `${file}.js`);
+      await fs.copy(`${dir}/worker.js.map`, `${file}.js.map`);
+      // Copy the current worker code as the "mjs" file as well.
+      // This is a temporary fix.
+      await fs.copy(`${dir}/worker.js`, `${file}.mjs`);
+      await fs.copy(`${dir}/worker.js.map`, `${file}.mjs.map`);
+
+      const noDomFile = `${targetDir}/amp-script-worker-nodom-${version}`;
+      // Same as above but for the nodom worker variant.
+      await fs.copy(`${dir}/worker.nodom.js`, `${noDomFile}.js`);
+      await fs.copy(`${dir}/worker.nodom.js.map`, `${noDomFile}.js.map`);
+      // Copy the current worker nodom code as the "mjs" file as well.
+      await fs.copy(`${dir}/worker.nodom.js`, `${noDomFile}.mjs`);
+      await fs.copy(`${dir}/worker.nodom.js.map`, `${noDomFile}.mjs.map`);
+    }
```

Comment:
This code is pretty substantial in size and complexity and should probably be moved into a separate function with its own JSDoc.

Reply:
done.